src/bin/lpython.cpp

```diff
@@ -804,6 +808,158 @@ int link_executable(const std::vector<std::string> &infiles,
 
 } // anonymous namespace
 
+#ifdef HAVE_BUILD_TO_WASM
+
+namespace wasm {
+
+#define INITIALIZE_VARS CompilerOptions compiler_options; \
+                        compiler_options.use_colors = true; \
+                        compiler_options.indent = true; \
+                        Allocator al(4*1024); \
+                        LFortran::LocationManager lm; \
+                        LFortran::diag::Diagnostics diagnostics; \
+                        lm.in_filename = "input";
+
+
+
+std::string out;
+
+extern "C" { // using extern "C" to prevent function name mangling
+
+EMSCRIPTEN_KEEPALIVE char* emit_ast_from_source(char *input) {
+    INITIALIZE_VARS;
+    lm.init_simple(input);
+    LFortran::Result<LFortran::LPython::AST::Module_t*> ast = LFortran::parse(al, input, diagnostics);
+    out = diagnostics.render(input, lm, compiler_options);
+    if (ast.ok) {
+        auto casted_ast = (LFortran::LPython::AST::ast_t*)ast.result;
+        out += LFortran::LPython::pickle_python(*casted_ast,
+            compiler_options.use_colors, compiler_options.indent);
+    }
+    return &out[0];
+}
+
+EMSCRIPTEN_KEEPALIVE char* emit_asr_from_source(char *input) {
+    INITIALIZE_VARS;
+    lm.init_simple(input);
+    LFortran::Result<LFortran::LPython::AST::Module_t*> ast = LFortran::parse(al, input, diagnostics);
+    out = diagnostics.render(input, lm, compiler_options);
+    if (ast.ok) {
+        auto casted_ast = (LFortran::LPython::AST::ast_t*)ast.result;
+        LFortran::Result<LFortran::ASR::TranslationUnit_t*>
+        asr = LFortran::LPython::python_ast_to_asr(al, *casted_ast, diagnostics, true,
+            compiler_options.disable_main, compiler_options.symtab_only, "input");
+        out += diagnostics.render(input, lm, compiler_options);
```

Comment:
**doubt:**
- Please, could someone possibly share if on Line `852` (and other similar lines), we shall just do `out = diagnostics.render(input, lm, compiler_options);` instead of `out += diagnostics.render(input, lm, compiler_options);`? 
    - Since, we did not execute `diagnostics.diagnostics.clear()` so all the previous diagnostics messages might still be present in the current diagnostics and therefore we can just assign the output of `diagnostics.render()` to `out` instead of append to `out`.

Reply:
I think the `diagnostics` variable itself contains the accumulation of all error messages, so I guess just `out = ...` is enough.

Reply:
Got it. I just pushed a commit with this change.